rt_sensorlevelanalysis.m

```diff
@@ -1,24 +1,17 @@
-function varargout = rt_sensorlevelanalysis(subj, cfg_preproc, type, mccaflag, saveflag)
+function varargout = rt_sensorlevelanalysis(subj, varargin)
 
 if nargin<1 || isempty(subj)
   subj = 'sub-004';
 end
 
-if nargin<2 || isempty(cfg_preproc)
-  cfg = [];
-else
-  cfg = cfg_preproc;
-end
-
-if nargin<3 || isempty(type)
-  type = 'timelock';
-end
+cfgpreproc = ft_getopt(varargin, 'cfgpreproc', []);
```

Comment:
Does the ft_getopt function in principle to the check was deleted in row 7-15, but with a flexible amount of var input for the function?

Reply:
Yes it does, but it is more flexible (does not rely on a fixed order of the input arguments). It comes at the cost that the syntax of calling the function is changed, because it requires 'key-value' pairs: 

output = functionname(input, 'key1', value1, 'key2', value2, ...etc)
the upside is that 
output = functionname(input, 'key2', value2, 'key1', value1) gives the same result as above,
and if you don't need key1, you can just omit it (rather then having to think about a certain number of empty arguments for instance

Reply:
Crystal clear, thanks a lot! And then by checking with ~isempty(cfgpreproc) it is possible to react flexibly to different cases that are defined by the variable input space.